go/test/endtoend/recovery/pitr/shardedpitr_test.go

```diff
@@ -0,0 +1,540 @@
+	// |  1 | prd-1  | 1597219030 |
+	// |  2 | prd-2  | 1597219030 |
+	// |  3 | prd-3  | 1597219043 |
+	// |  5 | prd-5  | 1597219045 |
+	// |  9 | prd-9  | 1597219130 |
+	// | 10 | prd-10 | 1597219130 |
+	// | 13 | prd-13 | 1597219141 |
+	// +----+--------+------------+
+	testTabletRecovery(t, bs0, "2m", restoreKS2Name, "-80", "INT64(7)")
+
+	// test the recovery with valid binlog_lookup_timeout for shard1 and getting the data till the restoreTime2
+	// 	mysql> select * from product;
+	// +----+--------+------------+
+	// | id | name   | created    |
+	// +----+--------+------------+
+	// |  4 | prd-4  | 1597219044 |
+	// |  6 | prd-6  | 1597219046 |
+	// |  7 | prd-7  | 1597219130 |
+	// |  8 | prd-8  | 1597219130 |
+	// | 11 | prd-11 | 1597219139 |
+	// | 12 | prd-12 | 1597219140 |
+	// +----+--------+------------+
+	testTabletRecovery(t, bs1, "2m", restoreKS2Name, "80-", "INT64(6)")
+
+	// test the recovery with timetorecover > (timestmap of last binlog event in binlog server)
+	createRestoreKeyspace(t, restoreTime3, restoreKS3Name)
+
+	// 	mysql> select * from product;
+	// +----+--------+------------+
+	// | id | name   | created    |
+	// +----+--------+------------+
+	// |  1 | prd-1  | 1597219030 |
+	// |  2 | prd-2  | 1597219030 |
+	// |  3 | prd-3  | 1597219043 |
+	// |  5 | prd-5  | 1597219045 |
+	// |  9 | prd-9  | 1597219130 |
+	// | 10 | prd-10 | 1597219130 |
+	// | 13 | prd-13 | 1597219141 |
+	// | 15 | prd-15 | 1597219142 |
+	// +----+--------+------------+
+	testTabletRecovery(t, bs0, "2m", restoreKS3Name, "-80", "INT64(8)")
+
+	// 	mysql> select * from product;
+	// +----+--------+------------+
+	// | id | name   | created    |
+	// +----+--------+------------+
+	// |  4 | prd-4  | 1597219044 |
+	// |  6 | prd-6  | 1597219046 |
+	// |  7 | prd-7  | 1597219130 |
+	// |  8 | prd-8  | 1597219130 |
+	// | 11 | prd-11 | 1597219139 |
+	// | 12 | prd-12 | 1597219140 |
+	// | 14 | prd-14 | 1597219142 |
+	// +----+--------+------------+
+	testTabletRecovery(t, bs1, "2m", restoreKS3Name, "80-", "INT64(7)")
+
+}
+
+func performResharding(t *testing.T) {
+	err := clusterInstance.VtctlclientProcess.ApplyVSchema(keyspaceName, vSchema)
+	require.NoError(t, err)
+
+	err = clusterInstance.VtctlProcess.ExecuteCommand("InitShardMaster", "-force", "ks/-80", shard0Master.Alias)
+	require.NoError(t, err)
+
+	err = clusterInstance.VtctlProcess.ExecuteCommand("InitShardMaster", "-force", "ks/80-", shard1Master.Alias)
+	require.NoError(t, err)
+
+	// we need to create the schema, and the worker will do data copying
+	for _, keyspaceShard := range []string{"ks/-80", "ks/80-"} {
+		err = clusterInstance.VtctlclientProcess.ExecuteCommand("CopySchemaShard", "ks/0", keyspaceShard)
+		require.NoError(t, err)
+	}
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("SplitClone", "ks", "0", "-80,80-")
+	require.NoError(t, err)
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("MigrateServedTypes", "ks/0", "rdonly")
```

Comment:
`MigrateServedTypes` are deprecated now, you need to use `SwtichRead` like [this](https://github.com/vitessio/vitess/blob/master/examples/local/203_switch_reads.sh). The old test still uses the old way, but the new ones should not use the deprecated ones.

Reply:
Used SwitchReads